src/main/java/ca/corefacility/bioinformatics/irida/processing/impl/AutomatedAnalysisFileProcessor.java

```diff
@@ -0,0 +1,156 @@
+
+			submission = submissionRepository.save(submission);
+
+			legacyFileProcessorCompatibility(submission, sequencingObject);
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public Boolean modifiesFile() {
+		return false;
+	}
+
+	/**
+	 * Get the {@link AnalysisSubmissionTemplate}s for a given {@link SequencingObject}.  Search up the {@link Sample}
+	 * and {@link Project}s the sequence belongs to.
+	 *
+	 * @param object the {@link SequencingObject}
+	 * @return the List of {@link AnalysisSubmissionTemplate}
+	 */
+	private List<AnalysisSubmissionTemplate> getAnalysisTemplates(SequencingObject object) {
+		List<AnalysisSubmissionTemplate> submissionTemplates = new ArrayList<>();
+
+		SampleSequencingObjectJoin sampleForSequencingObject = ssoRepository.getSampleForSequencingObject(object);
+
+		/*
+		 * This is something that should only ever happen in tests, but added
+		 * check with a warning
+		 */
+		if (sampleForSequencingObject != null) {
+			List<Join<Project, Sample>> projectForSample = psjRepository.getProjectForSample(
+					sampleForSequencingObject.getSubject());
+
+			for (Join<Project, Sample> j : projectForSample) {
+				List<AnalysisSubmissionTemplate> analysisSubmissionTemplatesForProject = analysisTemplateRepository.getAnalysisSubmissionTemplatesForProject(
+						j.getSubject());
+
+				submissionTemplates.addAll(analysisSubmissionTemplatesForProject);
+			}
+
+		} else {
+			logger.warn("Cannot find sample for sequencing object.  Not assembling");
+		}
+
+		return submissionTemplates;
+	}
+
+	/**
+	 * Do the work the old Assembly and SISTR file processors used to do and assign the assembly and SISTR result back
+	 * to the {@link SequencingObject}.  This will only do something if its an Assembly or SISTR analysis.
+	 *
+	 * @param submission       the {@link AnalysisSubmission} to check
+	 * @param sequencingObject the {@link SequencingObject} to apply results to.
+	 */
+	private void legacyFileProcessorCompatibility(AnalysisSubmission submission, SequencingObject sequencingObject) {
+		try {
+			IridaWorkflow assemblyWorkflow = workflowsService.getDefaultWorkflowByType(
+					BuiltInAnalysisTypes.ASSEMBLY_ANNOTATION);
+			IridaWorkflow sistrWorkflow = workflowsService.getDefaultWorkflowByType(BuiltInAnalysisTypes.SISTR_TYPING);
+
+			UUID assemblyWorkflowWorkflowIdentifier = assemblyWorkflow.getWorkflowIdentifier();
+			UUID sistrWorkflowWorkflowIdentifier = sistrWorkflow.getWorkflowIdentifier();
+
+			if (submission.getWorkflowId()
+					.equals(assemblyWorkflowWorkflowIdentifier)) {
+				// Associate the assembly submission with the seqobject
+				sequencingObject.setAutomatedAssembly(submission);
+
+				objectRepository.save(sequencingObject);
+			} else if (submission.getWorkflowId()
+					.equals(sistrWorkflowWorkflowIdentifier)) {
+				// Associate the sistr submission with the seqobject
+				sequencingObject.setSistrTyping(submission);
+
+				objectRepository.save(sequencingObject);
+			}
```

Comment:
An `else` here would hit for any pipeline that is not an `ASSEMBLY_ANNOTATION` or `SISTR_TYPING`, even if there's no problem.  I could add it at trace level, but it would generally be logging for things that are not problems.

Reply:
That's true. I did not think of it.

The update issue still occurs with `SISTR_TYPING` and `ASSEMBLY_ANNOTATION`. That is, once we make a new update to SISTR or ASSEMBLY, all automatically configured pipelines will fail to associated the results with the sequencing objects until someone goes through every project and updates them all. I was hoping to have an early warning of this.

Maybe you could also get all the existing `SISTR` and `ASSEMBLY_ANNOTATION` workflows using [getAllWorkflowsByType()](https://github.com/phac-nml/irida/blob/98357b5e7e4a8e55429d42a1e7118b25c5952a51/src/main/java/ca/corefacility/bioinformatics/irida/service/workflow/IridaWorkflowsService.java#L244). Then, have an `else if` statement that also checks to see if the `submission` analysis type is one of the previous `ASSEMBLY` or `SISTR` pipelines and logs a warning that automated SISTR and ASSEMBLY pipelines will not longer run as there has been an update?

Reply:
I have it as a task in #240 to disable old versions of automated pipelines when they've been updated.  I was thinking of doing something like this as a startup task and possibly having a warning on the project settings page if a pipeline is out of date.  I was planning on doing all of that as a later merge before it all gets rolled out.  I wanted to keep this one (relatively) small to just initiating and running the pipelines.  Does it work for you to leave it for now?

Reply:
Thanks. Yes this works for me. Having a warning on the project settings page will make it much more visible.